### ISSUE TITLE
"sudo make install" does place ctcsound.py under ~root/.local

### ISSUE THREAD

**Issue by quozl** [csound/csound#949]
When following BUILD.md for a Linux build, `sudo make install` does not put the Python module where it can be used by non-root users.  Instead of `/usr/local`, it is placed in `~root/.local`.

Impact is failure of `import ctcsound`.

Workaround is to copy the files to `/usr/local`, or copy `ctcsound.py` to current directory.

**Comment by fggp** [comment]
You can force the installation path for python modules by setting the
PYTHON_MODULE_INSTALL_DIR environment variable in your cmake command. For
example:

cmake -DPYTHON_MODULE_INSTALL_DIR=/usr/local/lib/python2.7/site-packaged
../csound

Then the sudo make install command will copy the files csnd6.py, _csnd6.so,
and ctcsound.py into that directory. If you want to use ctcsound.py in a
python3 session, you'll have to copy manually the ctcsound.py file in the
corresponding directory (on my system,
/usr/local/lib/python3.5/site-packages).

csnd6 and the python opcodes don't work with python3 and that's why cmake
actually detects a python2.x installation.

François

2018-04-24 1:41 GMT+02:00 James Cameron <notifications@github.com>:

**Comment by quozl** [comment]
Thanks.  Will you be patching `BUILD.md` or `CMakeLists.txt`?

At the moment, in `CMakeLists.txt`, `PYTHON_MODULE_INSTALL_DIR` defaults to `PYTHON_USER_SITE_PACKAGES`, yet the rest of Csound is installed to `/usr/local`.

I see the Debian package of Csound does do as you suggest;

```make
DEB_CMAKE_EXTRA_FLAGS = \
        -DBUILD_RELEASE=ON \
        ...
        -DPYTHON_MODULE_INSTALL_DIR=/usr/lib/python2.7/dist-packages \
        ...
```

**Comment by dvzrv** [comment]
It would be awesome to mention in BUILD.md, that python3 is actually not supported (yet).

**Comment by vlazzarini** [comment]
Well, it is supported by ctcsound.py; it is not supported by the SWIG-based wrapper.

**Comment by dvzrv** [comment]
@vlazzarini okay. I didn't know that. Is that documented somewhere?

**Comment by vlazzarini** [comment]
Not quite sure now where. Maybe we should ask the author: @fggp

**Comment by fggp** [comment]
ctcsound was introduced in Csound version 6.07. There is a short paragraph
about it in the API section of the Release notes for this version (
https://github.com/csound/csound/blob/develop/Release_Notes/Version_6.07).
Maybe it's time to write something in the manual 'Front Ends' section about
ctcsound. I will do that soon.

François


2018-05-13 1:07 GMT+02:00 vlazzarini <notifications@github.com>:

**Comment by jpffitch** [comment]
Is this still an issue or is it resolved?

**Comment by fggp** [comment]
Now ctcsound.py goes into /usr/local/lib/python2.x/dist-packages, and if Python3 is detected on the system, it goes too in /usr/local/lib/python3.x/dist-packages.

**Comment by fggp** [comment]
Closed #949

François

Le dim. 15 sept. 2019 à 21:21, John ffitch <notifications@github.com> a
écrit :

**Comment by Begasus** [comment]
This doesn't fix things for Haiku install, still need to set -DPYTHON_MODULE_INSTALL_DIR="$prefix/non-packaged/lib/python$PYTHON_VERSION/site-packages"
https://github.com/Begasus/haikuports/commit/b60b2ac9c607a0c7f9987bf90f01b0df2d5a75d6
